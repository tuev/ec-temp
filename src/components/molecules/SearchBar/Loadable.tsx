/**
 *
 * Asynchronously loads the component for SearchBar
 *
 */

import loadable from 'utils/loadable'

export default loadable(() => import('./index'))
