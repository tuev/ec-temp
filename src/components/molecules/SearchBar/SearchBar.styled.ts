/**
 *
 * Styled for SearchBar
 *
 */

import styled from 'styled-components'
import { TextField } from '@material-ui/core'
import { createMuiTheme } from '@material-ui/core/styles'
import { animated, SpringValue } from 'react-spring'

export const SearchInput = styled(animated(TextField)).attrs(
  ({ opacity, maxwidth }: { maxwidth: string; opacity: string }) => ({
    style: {
      opacity,
      maxWidth: maxwidth,
    },
  })
)<{
  maxwidth: SpringValue<string>
  opacity: SpringValue<number>
}>`
  border-radius: 50%;
`

export const SearchWrapper = styled.div`
  display: inline-block;
  position: relative;
  cursor: pointer;
`

export const theme = createMuiTheme({
  overrides: {
    MuiOutlinedInput: {
      root: {
        borderRadius: '24px',
      },
      input: {
        padding: '8px',
      },
    },
    MuiInputLabel: {
      outlined: {
        transform: 'translate(15px, 8px) scale(1)',
      },
    },
  },
})
