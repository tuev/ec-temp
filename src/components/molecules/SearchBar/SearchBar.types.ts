import { SpringValue } from 'react-spring'
export type SearchBarProps = {
  onSearch?: (inputString: string) => unknown
}

export interface ISearchBarAnimation {
  // TODO : check type for react-spring after 9.0 released

  [x: string]: any
  maxWidth: any
  opacity: any
}

export type SearchBarHook = (
  value: string,
  ref: React.RefObject<HTMLDivElement>
) => ISearchBarAnimation
