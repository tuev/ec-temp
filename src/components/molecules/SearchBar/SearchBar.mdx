# Stories mdx for SearchBar

import { Meta, Story } from '@storybook/addon-docs/blocks'
import * as stories from './SearchBar.stories.tsx'
import SearchBar from './SearchBar.tsx'

<Story name="basic" />