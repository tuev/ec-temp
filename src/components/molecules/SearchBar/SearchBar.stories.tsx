/**
 *
 *  Stories for SearchBar
 *
 */

import React from 'react'
import { action } from '@storybook/addon-actions'
import SearchBar from './SearchBar'

export default {
  component: SearchBar,
  includeStories: ['basic'],
  title: 'SearchBar',
}

export const basic = () => (
  <div>
    <SearchBar />
  </div>
)
