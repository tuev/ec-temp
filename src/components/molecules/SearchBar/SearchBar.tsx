/**
 *
 * SearchBar
 *
 */

import React, { FC, useRef, useState, useEffect, useCallback } from 'react'

import { Search } from '@material-ui/icons'
import Wrapper from 'components/atoms/Wrapper'
import { ThemeProvider } from '@material-ui/core'

import { SearchInput, theme, SearchWrapper } from './SearchBar.styled'
import useSearchBarTransition from './useSearchBarAnimation'
import { SearchBarProps } from './SearchBar.types'

const SearchBar: FC<SearchBarProps> = (props) => {
  const { onSearch } = props
  const [inputString, onChangeInput] = useState<string>('')
  const handleInputChange = useCallback(
    (e) => onChangeInput(e.target.value),
    []
  )
  useEffect(() => {
    if (onSearch) {
      onSearch(inputString)
    }
  }, [inputString, onSearch])

  const wrapperRef = useRef<HTMLDivElement>(null)
  const { maxWidth, opacity } = useSearchBarTransition(inputString, wrapperRef)

  return (
    <ThemeProvider theme={theme}>
      <SearchWrapper ref={wrapperRef} data-testid="search-ref">
        <SearchInput
          id="outlined-secondary"
          variant="outlined"
          color="secondary"
          maxwidth={maxWidth}
          opacity={opacity}
          onChange={handleInputChange}
          inputProps={{ 'data-testid': 'search-input' }}
        />
        <Wrapper position="absolute" top="8px" right="8px">
          <Search color="secondary" />
        </Wrapper>
      </SearchWrapper>
    </ThemeProvider>
  )
}

export default SearchBar
