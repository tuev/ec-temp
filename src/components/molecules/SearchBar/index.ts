/**
 *
 * Index for SearchBar
 *
 */

import SearchBar from './SearchBar'

export default SearchBar
