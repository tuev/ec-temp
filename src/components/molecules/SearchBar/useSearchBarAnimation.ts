import { useSpring } from 'react-spring'
import { useState, useCallback, useEffect } from 'react'
import { SearchBarHook } from './SearchBar.types'

const useSearchBarTransition: SearchBarHook = (value, ref) => {
  const [isSearch, toggleSearch] = useState(false)

  const showSearch = useCallback(() => toggleSearch(true), [])
  const hideSearch = useCallback(() => toggleSearch(false), [])

  const animtionStyle = useSpring({
    maxWidth: isSearch ? '195px' : '32px',
    opacity: isSearch ? 1 : 0,
  })

  const handleClick = useCallback(
    (e) => {
      if (ref.current && !ref.current.contains(e.target) && !value) {
        return hideSearch()
      }
      showSearch()
    },
    [hideSearch, ref, showSearch, value]
  )

  useEffect(() => {
    document.addEventListener('click', handleClick)
    return (): void => document.removeEventListener('click', handleClick)
  }, [handleClick])

  return animtionStyle
}

export default useSearchBarTransition
