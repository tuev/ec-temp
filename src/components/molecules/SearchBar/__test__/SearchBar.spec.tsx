/**
 *
 * Tests for SearchBar
 *
 */

import React from 'react'

import { render, fireEvent } from '@testing-library/react'
import SearchBar from '..'

describe('App SearchBar', () => {
  it('Test get default SearchBar no params ', () => {
    // test snapshot
    const wrapper = render(<SearchBar />)
    expect(wrapper).toMatchSnapshot()
  })

  it('Test get default SearchBar no event handler ', () => {
    // test snapshot
    const wrapper = render(<SearchBar />)
    const testString = 'this is search string'

    expect(wrapper).toMatchSnapshot()
    const inputComp = wrapper.getByTestId('search-input')
    fireEvent.change(inputComp, { target: { value: testString } })
  })

  it('Test get default SearchBar click to input ', () => {
    let result = ''
    const onSearchValue = jest
      .fn()
      .mockImplementation(value => (result = value))
    const testString = 'this is search string'
    // test snapshot
    const wrapper = render(<SearchBar onSearch={onSearchValue} />)
    expect(wrapper).toMatchSnapshot()
    const { asFragment } = wrapper
    const fragment = asFragment()
    const searchIcon = wrapper.getByTestId('search-ref')
    fireEvent.click(searchIcon)
    expect(asFragment()).toMatchDiffSnapshot()

    fireEvent.click(document)
    expect(asFragment()).toMatchDiffSnapshot()
  })

  it('Test SearchBar to input text', () => {
    let result = ''
    const onSearchValue = jest
      .fn()
      .mockImplementation(value => (result = value))
    const testString = 'this is search string'
    // test snapshot
    const wrapper = render(<SearchBar onSearch={onSearchValue} />)
    expect(wrapper).toMatchSnapshot()

    const inputComp = wrapper.getByTestId('search-input')
    fireEvent.change(inputComp, { target: { value: testString } })
    expect(onSearchValue).toBeCalled()
    expect(result).toEqual(testString)
  })
})
