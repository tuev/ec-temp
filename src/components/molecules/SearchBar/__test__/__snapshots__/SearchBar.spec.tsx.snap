// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App SearchBar Test SearchBar to input text 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    .c2 {
  position: absolute;
  top: 8px;
  right: 8px;
}

.c1 {
  border-radius: 50%;
}

.c0 {
  display: inline-block;
  position: relative;
  cursor: pointer;
}

<div>
      <div
        class="c0"
        data-testid="search-ref"
      >
        <div
          class="MuiFormControl-root MuiTextField-root c1"
          maxwidth="32px"
          opacity="0"
          style="opacity: 0; max-width: 32px;"
        >
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorSecondary MuiOutlinedInput-colorSecondary MuiInputBase-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input"
              data-testid="search-input"
              id="outlined-secondary"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-200 MuiOutlinedInput-notchedOutline"
              style="padding-left: 8px;"
            >
              <legend
                class="PrivateNotchedOutline-legend-201"
                style="width: 0.01px;"
              >
                <span>
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="c2"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-colorSecondary"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
            />
          </svg>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-AykKE igfrKc"
      data-testid="search-ref"
    >
      <div
        class="MuiFormControl-root MuiTextField-root sc-AykKD NTNrF"
        maxwidth="32px"
        opacity="0"
        style="opacity: 0; max-width: 32px;"
      >
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorSecondary MuiOutlinedInput-colorSecondary MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input"
            data-testid="search-input"
            id="outlined-secondary"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-200 MuiOutlinedInput-notchedOutline"
            style="padding-left: 8px;"
          >
            <legend
              class="PrivateNotchedOutline-legend-201"
              style="width: 0.01px;"
            >
              <span>
                ​
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="sc-AykKC jPsTgr"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-colorSecondary"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
          />
        </svg>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`App SearchBar Test get default SearchBar click to input  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    .c2 {
  position: absolute;
  top: 8px;
  right: 8px;
}

.c1 {
  border-radius: 50%;
}

.c0 {
  display: inline-block;
  position: relative;
  cursor: pointer;
}

<div>
      <div
        class="c0"
        data-testid="search-ref"
      >
        <div
          class="MuiFormControl-root MuiTextField-root c1"
          maxwidth="32px"
          opacity="0"
          style="opacity: 0; max-width: 32px;"
        >
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorSecondary MuiOutlinedInput-colorSecondary MuiInputBase-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input"
              data-testid="search-input"
              id="outlined-secondary"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-147 MuiOutlinedInput-notchedOutline"
              style="padding-left: 8px;"
            >
              <legend
                class="PrivateNotchedOutline-legend-148"
                style="width: 0.01px;"
              >
                <span>
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="c2"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-colorSecondary"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
            />
          </svg>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-AykKE igfrKc"
      data-testid="search-ref"
    >
      <div
        class="MuiFormControl-root MuiTextField-root sc-AykKD NTNrF"
        maxwidth="32px"
        opacity="0"
        style="opacity: 0; max-width: 32px;"
      >
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorSecondary MuiOutlinedInput-colorSecondary MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input"
            data-testid="search-input"
            id="outlined-secondary"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-147 MuiOutlinedInput-notchedOutline"
            style="padding-left: 8px;"
          >
            <legend
              class="PrivateNotchedOutline-legend-148"
              style="width: 0.01px;"
            >
              <span>
                ​
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="sc-AykKC jPsTgr"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-colorSecondary"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
          />
        </svg>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`App SearchBar Test get default SearchBar click to input  2`] = `
"Snapshot Diff:
  Comparing two different types of values. Expected object but received undefined."
`;

exports[`App SearchBar Test get default SearchBar click to input  3`] = `
"Snapshot Diff:
  Comparing two different types of values. Expected object but received undefined."
`;

exports[`App SearchBar Test get default SearchBar no event handler  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    .c2 {
  position: absolute;
  top: 8px;
  right: 8px;
}

.c1 {
  border-radius: 50%;
}

.c0 {
  display: inline-block;
  position: relative;
  cursor: pointer;
}

<div>
      <div
        class="c0"
        data-testid="search-ref"
      >
        <div
          class="MuiFormControl-root MuiTextField-root c1"
          maxwidth="32px"
          opacity="0"
          style="opacity: 0; max-width: 32px;"
        >
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorSecondary MuiOutlinedInput-colorSecondary MuiInputBase-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input"
              data-testid="search-input"
              id="outlined-secondary"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-94 MuiOutlinedInput-notchedOutline"
              style="padding-left: 8px;"
            >
              <legend
                class="PrivateNotchedOutline-legend-95"
                style="width: 0.01px;"
              >
                <span>
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="c2"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-colorSecondary"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
            />
          </svg>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-AykKE igfrKc"
      data-testid="search-ref"
    >
      <div
        class="MuiFormControl-root MuiTextField-root sc-AykKD NTNrF"
        maxwidth="32px"
        opacity="0"
        style="opacity: 0; max-width: 32px;"
      >
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorSecondary MuiOutlinedInput-colorSecondary MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input"
            data-testid="search-input"
            id="outlined-secondary"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-94 MuiOutlinedInput-notchedOutline"
            style="padding-left: 8px;"
          >
            <legend
              class="PrivateNotchedOutline-legend-95"
              style="width: 0.01px;"
            >
              <span>
                ​
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="sc-AykKC jPsTgr"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-colorSecondary"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
          />
        </svg>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`App SearchBar Test get default SearchBar no params  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c2 {
  position: absolute;
  top: 8px;
  right: 8px;
}

.c1 {
  border-radius: 50%;
}

.c0 {
  display: inline-block;
  position: relative;
  cursor: pointer;
}

<body>
    <div>
      <div
        class="c0"
        data-testid="search-ref"
      >
        <div
          class="MuiFormControl-root MuiTextField-root c1"
          maxwidth="32px"
          opacity="0"
          style="opacity: 0; max-width: 32px;"
        >
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorSecondary MuiOutlinedInput-colorSecondary MuiInputBase-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input"
              data-testid="search-input"
              id="outlined-secondary"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-41 MuiOutlinedInput-notchedOutline"
              style="padding-left: 8px;"
            >
              <legend
                class="PrivateNotchedOutline-legend-42"
                style="width: 0.01px;"
              >
                <span>
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="c2"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-colorSecondary"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
            />
          </svg>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-AykKE igfrKc"
      data-testid="search-ref"
    >
      <div
        class="MuiFormControl-root MuiTextField-root sc-AykKD NTNrF"
        maxwidth="32px"
        opacity="0"
        style="opacity: 0; max-width: 32px;"
      >
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorSecondary MuiOutlinedInput-colorSecondary MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input"
            data-testid="search-input"
            id="outlined-secondary"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-41 MuiOutlinedInput-notchedOutline"
            style="padding-left: 8px;"
          >
            <legend
              class="PrivateNotchedOutline-legend-42"
              style="width: 0.01px;"
            >
              <span>
                ​
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="sc-AykKC jPsTgr"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-colorSecondary"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
          />
        </svg>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
